theano/tensor/nnet/corr_gemm.c

```diff
@@ -172,18 +172,19 @@ PyArrayObject* corrMM(PyArrayObject* bottom,
     }
 
     // Create temporary columns
-    npy_intp col_dim[2];
-    col_dim[0] = (npy_intp)(nChannels * kW * kH);
-    col_dim[1] = (npy_intp)(topHeight * topWidth);
-    PyArrayObject* col = (PyArrayObject*)PyArray_EMPTY(2,
+    npy_intp col_dim[3];
+    col_dim[0] = (npy_intp)(batchSize);
```

Comment:
Maybe this will need to be aligned? I'm not sure of that, I think it is not mandatory to align, but maybe this can give a speed up.